<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1120</width>
    <height>921</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>3DViewer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="MyOpenGLWidget" name="openGLWidget">
      <property name="minimumSize">
       <size>
        <width>64</width>
        <height>64</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="widget" native="true">
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>Аффиновы преобразования:</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Вращение</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QPushButton" name="pushButton_rot_X_p">
                <property name="text">
                 <string>+X</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_rot_Y_p">
                <property name="text">
                 <string>+Y</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_rot_Z_p">
                <property name="text">
                 <string>+Z</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QPushButton" name="pushButton_rot_X_m">
                <property name="text">
                 <string>-X</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_rot_Y_m">
                <property name="text">
                 <string>-Y</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_rot_Z_m">
                <property name="text">
                 <string>-Z</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_11">
              <item>
               <widget class="QLabel" name="label_euler_x">
                <property name="text">
                 <string>x</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_euler_y">
                <property name="text">
                 <string>y</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_euler_z">
                <property name="text">
                 <string>z</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel" name="label_2">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="text">
               <string>Перемещение</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QPushButton" name="pushButton_trans_X_p">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <property name="text">
                 <string>+X</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_trans_Y_p">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <property name="text">
                 <string>+Y</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_trans_Z_p">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <property name="text">
                 <string>+Z</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_7">
              <item>
               <widget class="QPushButton" name="pushButton_trans_X_m">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <property name="text">
                 <string>-X</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_trans_Y_m">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <property name="text">
                 <string>-Y</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_trans_Z_m">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <property name="text">
                 <string>-Z</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel" name="label_3">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="text">
               <string>Масштаб</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QPushButton" name="pushButton_scale_p">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <property name="text">
                 <string>+</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_scale_m">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <property name="text">
                 <string>-</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_2">
           <property name="title">
            <string>Запись экрана:</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <widget class="QPushButton" name="pushButton_BMP">
                <property name="text">
                 <string>BMP</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_JPEG">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>JPEG</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_gif">
                <property name="text">
                 <string>GIF</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_3">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="focusPolicy">
            <enum>Qt::NoFocus</enum>
           </property>
           <property name="contextMenuPolicy">
            <enum>Qt::DefaultContextMenu</enum>
           </property>
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="title">
            <string>Цвет фона:</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <layout class="QGridLayout" name="gridLayout">
              <item row="0" column="4">
               <widget class="QPushButton" name="pushButton_bcgWhite">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="5">
               <widget class="QPushButton" name="pushButton_bcgGreen">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="9">
               <widget class="QPushButton" name="pushButton_bcgBrown">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QPushButton" name="pushButton_bcgPurple">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QPushButton" name="pushButton_bcgYellow">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="3">
               <widget class="QPushButton" name="pushButton_bcgBlack">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="8">
               <widget class="QPushButton" name="pushButton_bcgPink">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="6">
               <widget class="QPushButton" name="pushButton_bcgBlue">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QPushButton" name="pushButton_bcgGrey">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="7">
               <widget class="QPushButton" name="pushButton_bcgOrange">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_4">
           <property name="title">
            <string>Режимы отображения:</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QPushButton" name="pushButton_AutoRot">
              <property name="toolTip">
               <string>Автоповорот Старт/Стоп</string>
              </property>
              <property name="text">
               <string>Автоповорот (мировой)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_switchisProj">
              <property name="text">
               <string>Параллельная / Центральная</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_modeDraw">
              <property name="toolTip">
               <string>Режим отображения модели</string>
              </property>
              <property name="text">
               <string>Заливка</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_5">
           <property name="title">
            <string>Вершины:</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_9">
              <item>
               <widget class="QPushButton" name="pushButton_drawVerices">
                <property name="text">
                 <string>Тип</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_VSize_p">
                <property name="text">
                 <string>+</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_VSize_m">
                <property name="text">
                 <string>-</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QGridLayout" name="gridLayout_3">
              <item row="0" column="1">
               <widget class="QPushButton" name="pushButton_vertexYellow">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="4">
               <widget class="QPushButton" name="pushButton_vertexWhite">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="8">
               <widget class="QPushButton" name="pushButton_vertexPink">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="9">
               <widget class="QPushButton" name="pushButton_vertexBrown">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QPushButton" name="pushButton_vertexPurple">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="3">
               <widget class="QPushButton" name="pushButton_vertexBlack">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QPushButton" name="pushButton_vertexGrey">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="6">
               <widget class="QPushButton" name="pushButton_vertexBlue">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="5">
               <widget class="QPushButton" name="pushButton_vertexGreen">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="7">
               <widget class="QPushButton" name="pushButton_vertexOrange">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_6">
           <property name="title">
            <string>Линии:</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_10">
              <item>
               <widget class="QPushButton" name="pushButton_dashedLine">
                <property name="text">
                 <string>Тип</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_lineSize_p">
                <property name="text">
                 <string>+</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_lineSize_m">
                <property name="text">
                 <string>-</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="0" column="8">
               <widget class="QPushButton" name="pushButton_edgePink">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="9">
               <widget class="QPushButton" name="pushButton_edgeBrown">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QPushButton" name="pushButton_edgePurple">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="4">
               <widget class="QPushButton" name="pushButton_edgeWhite">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QPushButton" name="pushButton_edgeYellow">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="6">
               <widget class="QPushButton" name="pushButton_edgeBlue">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QPushButton" name="pushButton_edgeGrey">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="3">
               <widget class="QPushButton" name="pushButton_edgeBlack">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="5">
               <widget class="QPushButton" name="pushButton_edgeGreen">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="7">
               <widget class="QPushButton" name="pushButton_edgeOrange">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1120</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="action_openFile"/>
    <addaction name="action_saveSettings"/>
    <addaction name="action_close"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action_openFile">
   <property name="text">
    <string>Открыть файл</string>
   </property>
  </action>
  <action name="action_close">
   <property name="text">
    <string>Закрыть программу</string>
   </property>
  </action>
  <action name="action_saveSettings">
   <property name="text">
    <string>Сохранить настройки</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyOpenGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>myopenglwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
